### PR TITLE
Make proxy use roundrobin to choose replica

### LINKED ISSUE #17055: [Bug]: Proxy are using the same replica all the time
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

Even when we have multiple replicas, proxy uses one replica all the time.

### Expected Behavior

_No response_

### Steps To Reproduce

_No response_

### Milvus Log

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by XuanYang-cn**
Fixes: #17055

Signed-off-by: yangxuan <xuan.yang@zilliz.com>

/kind bug

### CODE REVIEW

File: internal/proxy/task_policies.go [hunk 1]
```diff
@@ -40,23 +40,65 @@ var (
 	errInvalidShardLeaders = errors.New("Invalid shard leader")
 )
 
-func roundRobinPolicy(ctx context.Context, getQueryNodePolicy getQueryNodePolicy, query func(UniqueID, types.QueryNode) error, leaders *querypb.ShardLeadersList) error {
+type queryNode struct {
+	nodID   UniqueID
```
Comment (yah01) [inline internal/proxy/task_policies.go:44]: typo: nodID -> nodeID

File: internal/proxy/task_query.go [hunk 2]
```diff
@@ -344,12 +345,13 @@ func (t *queryTask) PostExecute(ctx context.Context) error {
 	return nil
 }
 
-func (t *queryTask) queryShard(ctx context.Context, leaders *querypb.ShardLeadersList) error {
+// func (t *queryTask) queryShard(ctx context.Context, leaders *querypb.ShardLeadersList) error {
```
Comment (yah01) [inline internal/proxy/task_query.go:348]: remove

File: internal/proxy/task_policies.go [hunk 3]
```diff
@@ -40,23 +40,65 @@ var (
...
+			queue <- leader
+		}
+
+		rearrangedLeaders := roundRobin(queue)
+		close(queue)
+		for leader := range rearrangedLeaders {
+			result = append(result, leader)
+		}
+
+		shardsLeaders[channelID] = result
```
Comment (yah01) [inline internal/proxy/task_policies.go:71]: just:
```
shardsLeaders[channelID] = append(leaders[1:], leaders[0])
```

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**